timelock-agent/src/main/java/com/palantir/timelock/config/DatabaseTsBoundPersisterConfiguration.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.config;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+
+@JsonDeserialize(as = ImmutableDatabaseTsBoundPersisterConfiguration.class)
+@JsonSerialize(as = ImmutableDatabaseTsBoundPersisterConfiguration.class)
+@Value.Immutable
+public abstract class DatabaseTsBoundPersisterConfiguration implements TsBoundPersisterConfiguration {
+
+    @JsonProperty("key-value-service")
+    public abstract KeyValueServiceConfig keyValueServiceConfig();
```

Comment:
@tpetracca I tried removing C* dependency here:
```
public abstract class DatabaseTsBoundPersisterConfiguration implements TsBoundPersisterConfiguration {

    @JsonProperty("key-value-service")
    public abstract KeyValueServiceConfig keyValueServiceConfig();

    @Value.Check
    public void check() {
        Preconditions.checkArgument(DbKeyValueServiceConfig.class.isInstance(keyValueServiceConfig()));
    }

}
```

This needs a dependency on ``atlasdb-dbkvs`` in timelock-agent..I would be unhappy if I added the dependency here and also unhappy if this is not a config check but a runtime error (if C* kvs is configured for TsPersistence). 

Would be happy to implement if you have a better solution here.

Reply:
thinking further, I can add the check to CassandraAtlasDbFactory too.. slightly cleaner, but stil not a config level check, going with that for now.